libsolidity/ast/Types.cpp

```diff
@@ -294,35 +294,6 @@ Type const* Type::commonType(Type const* _a, Type const* _b)
 		return nullptr;
 }
 
-char const* Type::categoryName(Type::Category _category)
-{
-	switch (_category)
-	{
-	case Category::Address: return "address";
-	case Category::Integer: return "integer";
-	case Category::RationalNumber: return "rational number literal";
-	case Category::StringLiteral: return "string literal";
-	case Category::Bool: return "boolean";
-	case Category::FixedPoint: return "fixed-point number";
-	case Category::Array: return "array";
-	case Category::ArraySlice: return "array slice";
```

Comment:
Shouldn't we have tests for all types? It seems that it is missing some cases, e.g. arrays, boolean.

Reply:
Technically we should for full coverage, but the gain from that would be very marginal since, at least in the case that triggers the message, they all go though the same code path and it's only the message text that used to differ (and now it's even identical).

I did add more of them back when I introduced `categoryName()` but even that was mostly to see the effect of my changes in context. I think we have enough of them now.